go/saltpack/specs/saltpack_encryption.md

```diff
@@ -113,30 +120,81 @@ A recipient pair is a two-element list:
 
 - The **recipient public key** is the recipient's long-term NaCl public
   encryption key. This field may be null, when the recipients are anonymous.
-- The **payload key box** is a NaCl box encrypted with the recipient's public
-  key and the ephemeral private key. It contains the **payload key**, a 32-byte
-  NaCl symmetric encryption key used to encrypt the **sender public key
-  secretbox** above that the **payload secretbox** in each payload packet
-  below. The sender generates a random payload key for each message.
-
-Recipients should compute the ephemeral shared secret and then try to open each
-of the **payload key boxes**. In the NaCl interface, computing a shared secret
-is done with the [`crypto_box_beforenm`](http://nacl.cr.yp.to/box.html)
-function, which does a Curve25519 multiplication and an HSalsa20 key
-derivation, to avoid repeating those steps in
-[`crypto_box_open`](http://nacl.cr.yp.to/box.html). After obtaining the
-**payload key**, clients open the **sender public key secretbox** and obtain
-the sender's public key. Both keys are used to decrypt payload packets below.
-
-Encrypting the sender's long-term public key allows Alice to stay anonymous to
-Mallory. If Alice wants to be anonymous to Bob as well, she can reuse the
-ephemeral key as her sender key. When the ephemeral key and the sender key are
-the same, clients may indicate that a message is "intentionally anonymous" as
-opposed to "from an unknown sender".
+- The **payload key box** is a NaCl box containing a copy of the **payload
+  key**. It's encrypted with the recipient's public key and the ephemeral
+  private key. (See [Nonces](#nonces).)
+
+### Generating a Header Packet
+
+When composing a message, the sender follows these steps to generate the
+header:
+
+1. Generate a random 32-byte **payload key**.
+2. Generate a random ephemeral keypair, using
+   [`crypto_box_keypair`](http://nacl.cr.yp.to/box.html).
+3. Encrypt the sender's long-term public key using
+   [`crypto_secretbox`](http://nacl.cr.yp.to/secretbox.html) with the **payload
+   key**, to create the **sender secretbox**. (See [Nonces](#nonces).)
```

Comment:
Instead say `(See [Nonces](#nonces) below for a description of which nonce to used)`.